object/map.js

```diff
@@ -0,0 +1,11 @@
+
+if (!Object.prototype.map) {
+  Object.defineProperty(Object.prototype, 'map', {
+    value: function(callback) {
+      var object = this
```

Comment:
Right. I'm wondering. Should we include a test that ensures the original value of the object doesn't get mutated? I don't think that's desired behavior.